scripts/azureml-assets/azureml/assets/model/registry_utils.py

```diff
@@ -4,29 +4,94 @@
 
 import copy
 import azureml.assets as assets
-from typing import Tuple
-from azure.ai.ml import load_model, MLClient
+from typing import Tuple, Union
+from azure.ai.ml import load_model, load_data, MLClient, operations as ops
 from azure.ai.ml._utils._registry_utils import get_asset_body_for_registry_storage
 from azureml.assets.util import logger
+from azureml.assets.config import AssetType, Config, DataConfig, ModelConfig
 from azureml.assets.util.util import resolve_from_file_for_asset
 from azureml.assets.config import PathType
 from azureml.assets.model.download_utils import CopyUpdater, copy_azure_artifacts, download_git_model
 from azureml.assets.deployment_config import AssetVersionUpdate
+from pathlib import Path
 
 
 class RegistryUtils:
     """Registry utils."""
 
     RETRY_COUNT = 3
 
-    def get_registry_data_reference(model_name: str, model_version: str, ml_client: MLClient) -> Tuple[str, str]:
+    # Note: taken from publish_utils, but causes circular import if importing it
```

Comment:
thinking we can keep this here and remove the one from publish_utils

Reply:
Do you want to do that in this PR, checking for and updating any references to the functions?

Reply:
Updated this PR, thanks!